tests/unit/package_managers/pip/test_package_distributions.py

```diff
@@ -335,3 +337,190 @@ def test_process_package_distributions_noncanonical_version(
     assert artifacts[0].package_type == "sdist"
     assert artifacts[0].version == requested_version
     assert all(w.version == requested_version for w in artifacts[1:])
+
+
+class TestWheelsFilter:
+    @pytest.mark.parametrize(
+        "filter_kwargs,expected",
+        [
+            pytest.param(
+                {},
+                {
+                    "packages": None,
+                    "arch": {"x86_64"},
+                    "os": {"linux"},
+                    "py_version": None,
+                    "py_impl": {"cp"},
+                    "abi": None,
+                    "platform_regex": None,
+                },
+                id="default_filters",
+            ),
+            pytest.param(
+                {
+                    "packages": "numpy,pandas",
+                    "arch": "x86_64,aarch64",
+                    "os": "linux,macos",
+                    "py_version": 312,
+                    "py_impl": "cp,pp",
+                },
+                {
+                    "packages": {"numpy", "pandas"},
+                    "arch": {"x86_64", "aarch64"},
+                    "os": {"linux", "macos"},
+                    "py_version": 312,
+                    "py_impl": {"cp", "pp"},
+                },
+                id="custom_filters",
+            ),
+            pytest.param(
+                {
+                    "packages": ":all:",
+                    "arch": ":all:",
+                    "os": ":all:",
+                    "py_impl": ":all:",
+                    "abi": ":all:",
+                },
+                {
+                    "packages": None,
+                    "arch": None,
+                    "os": None,
+                    "py_impl": None,
+                    "abi": None,
+                },
```

Comment:
nitpick: You might also replace this with `{ key: None for key in ["packages", "arch", "os", "py_impl", "abi"] },` to fit it on a single line.